UnityEngineAnalyzer/UnityEngineAnalyzer/DiagnosticIDs.cs

```diff
@@ -13,12 +13,21 @@ public static class DiagnosticIDs
         public const string UnsealedDerivedClass = "UEA0008";
         public const string InvokeFunctionMissing = "UEA0009";
         public const string DoNotUseStateNameInAnimator = "UEA0010";
+        public const string DoNotUseMaterialNameInMaterial = "UEA0011";
+        public const string DoNotUseCameraMainInUpdate = "UEA0012";
+        public const string DoNotGCAllocInUpdate = "UEA0013";
 
         //NOTES: These should probably be on their own analyzer - as they are not specific to Unity
         public const string DoNotUseRemoting = "AOT0001";
         public const string DoNotUseReflectionEmit = "AOT0002";
         public const string TypeGetType = "AOT0003";
+        public const string StructShouldImplementIEquatable = "AOT0004";
+        public const string StructShouldOverrideEquals = "AOT0005";
+        public const string StructShouldOverrideGetHashCode = "AOT0006";
+        public const string DoNotBoxWhenInvoke = "AOT0007";
+        public const string DoNoUseEnumTypeParameter = "AOT0008";
+        public const string ShouldCacheDelegate = "AOT0009";
```

Comment:
I am assuming that this is a performance / GC problem not an AOT problem. Is the the correct category?